File: problems/Simplify Path/simplifyPath.cs
public class Solution {
    public string SimplifyPath(string path) {
        var stack = new Stack<string>();
        var store = path.Trim('/').Split('/');

        foreach (var item in store) {
            if ("." == item || string.IsNullOrEmpty(item)) {
                continue;
            } else if (".." == item) {
                if (0 < stack.Count) {
                    stack.Pop();
                }
            } else {
                stack.Push(item);
            }
        }

        return '/' + string.Join("/", stack.Reverse());
    }
}

File: problems/Simplify Path/simplifyPath.java
class Solution {
    public String simplifyPath(String path) {
        StringBuilder sb = new StringBuilder();
        Stack<String> store = new Stack<>();
        String[] components = path.split("/");
        
        for (String component : components) {
            if (component.equals(".") || component.isEmpty()) {
                continue;
            } else if (component.equals("..")) {
                if (!store.isEmpty()) {
                    store.pop();
                }
            } else {
                store.add(component);
            }
        }

        while (!store.isEmpty()) {
            sb.insert(0, store.pop());
            sb.insert(0, "/");
        }

        return 0 < sb.length() ? sb.toString() : "/";
    }
}
